### PR TITLE
feat(dyn-sampling): Switch to new feature  flag multiplexer in projectconfig

### PR DESCRIPTION
**Pull request by andriisoldatenko**
This PR switch to new feature flag multiplexer
in projectconfig.

### CODE REVIEW

**Review by ahmedetefy — APPROVED** [review]

File: tests/sentry/api/endpoints/test_relay_projectconfigs.py [hunk 1]
```diff
@@ -169,7 +169,12 @@ def test_relays_dyamic_sampling(client, call_endpoint, default_project, dyn_samp
     """
     default_project.update_option("sentry:dynamic_sampling", dyn_sampling_data())
 
-    with Feature({"organizations:server-side-sampling": True}):
+    with Feature(
+        {
+            "organizations:server-side-sampling": True,
```
Comment (jjbayer) [inline tests/sentry/api/endpoints/test_relay_projectconfigs.py:174]: I think we can now even remove `server-side-sampling` here now, right?

Comment (andriisoldatenko) [reply tests/sentry/api/endpoints/test_relay_projectconfigs.py:174]: correct but we decided with @matejminar and @ahmedetefy that we can do it in follow up PR.

Comment (jjbayer) [reply tests/sentry/api/endpoints/test_relay_projectconfigs.py:174]: Ah, the multiplexer still checks server-side-sampling, sorry, did not realize that.

File: src/sentry/relay/config/__init__.py [hunk 2]
```diff
@@ -155,20 +156,13 @@ def get_project_config(project, full_config=True, project_keys=None):
...
 def get_dynamic_sampling_config(project) -> Optional[Mapping[str, Any]]:
-    allow_server_side_sampling = features.has(
-        "organizations:server-side-sampling",
-        project.organization,
-    )
-    allow_dynamic_sampling = features.has(
-        "organizations:dynamic-sampling",
-        project.organization,
-    )
+    feature_multiplexer = DynamicSamplingFeatureMultiplexer(project, None)
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:159]: Not part of this PR, but we could probably use `features.batch_has` in the multiplexer's `__init__`.

**Review by jjbayer — APPROVED** [review]

File: src/sentry/relay/config/__init__.py (hunk 2 above)
Comment (andriisoldatenko) [reply src/sentry/relay/config/__init__.py:159]: yeah good idea.
